src/client.ts

```diff
@@ -155,17 +154,22 @@ class VeracodeClient {
     try {
       const result = await findingsRequest;
       response = JSON.parse(result.body);
-      return {
+      const toReturn = {
         items: response._embedded?.findings || [],
         nextUri: response._links.next?.href,
       };
+      if (toReturn.nextUri && toReturn.items.length !== FINDINGS_PAGE_SIZE) {
```

Comment:
[veracode findings api](https://app.swaggerhub.com/apis/Veracode/veracode-findings_api_specification/2.0#/Application%20findings%20information./getFindingsUsingGET) does not advertise a pageSize option. However, it arrives on the `nextUri` when performing pagination. Default page size is 20. For some multi-thousand issue applications, this becomes painfully slow, and can lead to [rate-limiting](https://docs.veracode.com/r/About_API_Rate_Limiting). On [other routes](https://app.swaggerhub.com/apis/Veracode/veracode-findings_api_specification/2.0#/CWE%20API%20information/getCategoriesUsingGET), Veracode does have a pageSize option with a maximum of 500. When you specify greater than this maximum, you will still get a `200` response, but you will be left with only 500 items and a `nextUri` that actually **acts as though the # of items returned was what was requested**. We should report this issue to them to avoid potential for lost data. This check is here in hopes of catching this issue on very large scans, but it is by no means exhaustive. I believe the substantial speed increase of the integration (almost an order of magnitude faster) and avoiding getting rate-limited due to a meager page size of 20 is worth it however.

Reply:
IMO this is is really our only avenue to avoid significant rate-limiting in the face of a substantially large code base in veracode. Note also that every finding has to be iterated over regardless of status (the only findings we can skip over are ones that **do not violate the policy of the application it belongs to**) . There is no way to derive (let alone specify) a `modified_since` value. We could omit all “Mitigated Findings” from the graph, but the fact remains that everything must be requested from the API. It is my understanding however, that the API gives the `current set` of findings for the application from the most recent set of scans.

Reply:
Thanks for investigating this. I'd agree that we'll probably need a bigger page size and hopefully they can fully support it. In the if why does the condition need to be `toReturn.nextUri && toReturn.items.length !== FINDINGS_PAGE_SIZE)`. Is it not possible for there to be mismatching lengths when there isn't a `nextUri`.

Also, could it be possible on the last page for the number of items to be less than the page size?

Reply:
RE: `Is it not possible for there to be mismatching lengths when there isn't a nextUri?`

Yes it is possible. If there is no `nextUri`, we have to assume we are on the last page. However, if there were `>= ACTUAL_MAXIMUM + 1 where FINDINGS_PAGE_SIZE > ACTUAL_MAXIMUM` items and we were falling victim to this bug, we would be on the last page according to the API, but would be missing data. I don't believe there is a good way around this. We will be reporting this to veracode after i gather a few examples of this behaviour.